packages/state/source/wallets/wallet-repository-clone.ts

```diff
@@ -77,16 +78,18 @@ export class WalletRepositoryClone extends WalletRepository implements Contracts
 		}
 	}
 
-	public getDirtyWallets(): ReadonlyArray<Contracts.State.Wallet> {
-		return this.getIndex(Contracts.State.WalletIndexes.Addresses)
-			.values()
-			.filter((walletHolder) => walletHolder.isChanged());
+	public getDirtyWallets(): IterableIterator<Contracts.State.Wallet> {
+		return this.#dirtyWallets.values();
+	}
+
+	public setDirtyWallet(wallet: Contracts.State.Wallet): void {
+		this.#dirtyWallets.add(wallet);
 	}
 
 	public commitChanges(): void {
 		// Merge clones to originals
-		for (const wallet of this.getDirtyWallets()) {
-			wallet.commitChanges();
+		for (const wallet of this.#dirtyWallets.values()) {
```

Comment:
Double check if we should clear the set after committing changes

Reply:
WalletRepository shouldn’t be used after changes are committed. We process block only once. After commit new WalletRepository is created.